pip/req/req_install.py

```diff
@@ -421,6 +421,14 @@ def run_egg_info(self):
                     self.pkg_info()["Version"],
                 ]))
             self._correct_build_location()
+        else:
+            metadata_name = canonicalize_name(self.pkg_info()["Name"])
+            if self.req.project_name.lower() != metadata_name:
```

Comment:
Well `self.req.project_name.lower()` is equivalent to the canonical name but I agree this is not clear (and only true for the current definition of `canonicalize_name`). I'll `canonicalize_name` both terms.